xbmc/Application.cpp

```diff
@@ -3216,6 +3216,10 @@ void CApplication::OnPlayBackStarted(const CFileItem &file)
 {
   CLog::LogF(LOGDEBUG,"CApplication::OnPlayBackStarted");
 
+  // check if VideoPlayer should set file item stream details from its current streams
+  if (file.GetProperty("get_stream_details_from_player").asBoolean())
```

Comment:
Is `get_stream_details_from_player` used anywhere else? It's better to use defines instead of strings because the compiler can catch spelling errors.

If this property is specific to VideoPlayer, then VideoPlayer should define it. You can create a new header, possibly `VideoPlayerDefines.h`, like [this](https://github.com/garbear/xbmc/blob/retroplayer-shaders/xbmc/games/dialogs/DialogGameDefines.h), which can scale to handle future properties.

If there's no real clear place for a define, just leave it as a string and deal with tears from baby jesus.

Reply:
> Is get_stream_details_from_player used anywhere else?

No. Currently this property is made specially for plugins, so that they can set it to ListItem for setResolvenUrl (if a internet streaming video plugin wants to update file item stream details from VP current streams).

> If this property is specific to VideoPlayer

Even if we add this to other players, this check will leave in CApplication and there will be the only place where this constant is used.

If in the future it will be used somewhere else, then, of course, it should be done with define. If you don't mind, I would leave it as it is for now.

Reply:
coolio